packages/evolution-backend/src/services/routing/RouteCalculationFromTransition.ts

```diff
@@ -0,0 +1,125 @@
+import { URL } from 'url';
+
+// TODO Handle token expiration. For now though, the evolution app will restart more often than the token expiration
+let token: string | undefined = undefined;
+
+const getTransitionUrl = (transitionUrl: string) =>
+    new URL(transitionUrl.startsWith('http') ? transitionUrl : `http://${transitionUrl}`);
+
+const getToken = async () => {
+    if (token !== undefined) {
+        return token;
+    }
+
+    const transitionUrlObj = getTransitionUrl(projectConfig.transitionApi!.url);
+    transitionUrlObj.pathname = '/token';
+    const response = await fetch(transitionUrlObj.toString(), {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+            usernameOrEmail: projectConfig.transitionApi!.username,
+            password: projectConfig.transitionApi!.password
+        })
+    });
+
+    // FIXME Every request will try to get a token, even if the previous one failed. Should we cache the error and not retry? Or retry only after x time?
+    if (response.status !== 200) {
+        throw new Error(`Cannot get token from transition: ${response.status}`);
+    }
+
+    const tokenResponse = await response.text();
+    token = tokenResponse;
+    return token;
+};
+
+/**
+ * Calculate the routes for a list of modes
+ * @param modes The list of modes to calculate the routes for
+ * @param parameters The parameters for the route calculation
+ * @param withGeojson Whether to include the GeoJSON in the response
+ */
+export default async (
+    modes: RoutingOrTransitMode[],
+    parameters: RouteCalculationParameter
+): Promise<RoutingResultByMode> => {
+    const transitionUrl = projectConfig.transitionApi?.url;
+    if (transitionUrl === undefined) {
+        throw new Error('Transition URL not set in project config');
+    }
+
+    const transitionUrlObj = getTransitionUrl(transitionUrl);
+    transitionUrlObj.pathname = '/api/v1/route';
+    transitionUrlObj.searchParams.append('withGeojson', parameters.withGeojson === true ? 'true' : 'false');
+
+    try {
+        const response = await fetch(transitionUrlObj.toString(), {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+                Authorization: `Bearer ${await getToken()}`
+            },
+            body: JSON.stringify({
+                routingModes: modes,
+                originGeojson: parameters.origin,
+                destinationGeojson: parameters.destination,
+                departureTimeSecondsSinceMidnight: parameters.departureSecondsSinceMidnight,
+                scenarioId: parameters.transitScenario
+            })
+        });
+
+        if (response.status !== 200) {
+            throw new Error(`Unsuccessful response code from transition: ${response.status}`);
+        }
+        const routingResponse = await response.json();
+        const travelTimeByMode: RoutingResultByMode = {};
```

Comment:
I don't like the name of the travelTimeByMode variable. It's not just a travelTime that we return but a full route.